### ISSUE TITLE
Test for WCAG success criterion 1.1.1: Non-text Content

### ISSUE THREAD

**Issue by hannolans** [quailjs/quail#205]
http://www.w3.org/TR/UNDERSTANDING-WCAG20/text-equiv-all.html

Non-text Content: All non-text content that is presented to the user has a text alternative that serves the equivalent purpose.

The intent of this Success Criterion is to make information conveyed by non-text content accessible through the use of a text alternative. Text alternatives are a primary way for making information accessible because they can be rendered through any sensory modality (for example, visual, auditory or tactile) to match the needs of the user. Providing text alternatives allows the information to be rendered in a variety of ways by a variety of user agents. For example, a person who cannot see a picture can have the text alternative read aloud using synthesized speech. A person who cannot hear an audio file can have the text alternative displayed so that he or she can read it. In the future, text alternatives will also allow information to be more easily translated into sign language or into a simpler form of the same language.
## Applicable

Non-text content
- Situation A: If a short description can serve the same purpose and present the same information as the non-text content
- Situation B: If a short description can not serve the same purpose and present the same information as the non-text content (e.g., a chart, map or diagram)
- Situation C: If non-text content is a control or accepts user input
- Situation D: If non-text content is time-based media (including live video-only and live audio-only); a test or exercise that would be invalid if presented in text; or primarily intended to create a specific sensory experience (works of visual art)
- Situation E: If non-text content is a CAPTCHA
- Situation F: If the non-text content should be ignored by assistive technology
## Selector
- IMG etc
## Failures
- [ ] F3: Using CSS to include images that convey important information
- [ ] F13: Having a text alternative that does not include information that is conveyed by color differences in the image
- [ ] F20: Not updating text alternatives when changes to non-text content occur
- [ ] F30: Using text alternatives that are not alternatives (e.g., filenames or placeholder text)
- [ ] F38: Not marking up decorative images in HTML in a way that allows assistive technology to ignore them
- [ ] F39: Providing a text alternative that is not null (e.g., alt="spacer" or alt="image") for images that should be ignored by assistive technology
- [ ] F65: Omitting the alt attribute or text alternative on img elements, area elements, and input elements of type "image"
- [ ]  F67: Providing long descriptions for non-text content that does not serve the same purpose or does not present the same information
- [ ] F71: Using text look-alikes to represent text without providing a text alternative
- [ ] F72: Using ASCII art without providing a text alternative
